antivirus.py

```diff
@@ -107,49 +139,104 @@ def start(self) -> None:
     def execute(self, request: ServiceRequest) -> None:
         request.result = Result()
         max_workers = len(self.hosts)
-        result_sections = []
+        av_version_result_sections: List[ResultSection] = []
+        av_hit_result_sections: List[AvHitSection] = []
+        request.deep_scan = True
         with ThreadPoolExecutor(max_workers=max_workers) as executor:
             # Submit the file to each of the hosts
             futures = [
-                executor.submit(self._scan_file, host, request.file_name, request.file_contents)
+                executor.submit(self._scan_file, host, request.file_name, request.file_contents, request.deep_scan)
                 for host in self.hosts if not host.sleeping
             ]
             for future in futures:
-                result, host = future.result()
-                result_section = AntiVirus._parse_result(result, host.method, host.name) if result is not None else None
-                if result_section is not None:
-                    result_sections.append(result_section)
-        AntiVirus._gather_results(self.hosts, result_sections, request.result)
+                result, version, host = future.result()

```

Comment:
Possible improvement (not that critical):
Iterate on futures based on status so completed threads get processed first (and potentially popped from collection so another pass can be given to longer running threads at a later point).

The scenario in mind is if an AV finished scanning before another, it doesn't have to wait on the slower AV to process results just because of precedence when the threads were created.